### PR TITLE
Added "hosted by" filter

### PR DESCRIPTION
**Pull request by etnKlendathu**

### CODE REVIEW

File: server/test/resolve.cpp [hunk 1]
```diff
@@ -1108,3 +1108,106 @@ TEST_CASE("Resolve by ip address vm")
         FAIL(ex.what());
     }
 }
+
+TEST_CASE("Resolve by hosted by")
+{
+    try {
+        fty::SampleDb db(R"(
```
Comment (perrettecl) [inline server/test/resolve.cpp:1115]: @EatonEKia Do you think the test is covering all your case ?

**Review by perrettecl — APPROVED** [review]

File: server/test/resolve.cpp (hunk 1 above)
Comment (etnKlendathu) [reply server/test/resolve.cpp:1115]: No, this test just for basic functionality....
I don't know (and can't imagine) all corner cases for now... if you have something, I would be glad to add it. :)

Comment (etnKlendathu) [reply server/test/resolve.cpp:1115]: I don't think so... this is just basic test which I can imagine... 
@EatonEKia Also I want to request more test cases.

Comment (EatonEKia) [reply server/test/resolve.cpp:1115]: He will cover the VMware case where VMs are hosted on an ESXi. You will have same cases with Microsoft (microsoft.hyperv.hosts.vm) and Nutanix (nutanix.node.hosts.vm).

Comment (etnKlendathu) [reply server/test/resolve.cpp:1115]: But difference only in link type or something else?

Comment (EatonEKia) [reply server/test/resolve.cpp:1115]: No, you're absolutly righ.
